nmt-python-advanced-finetune multilingual-nmt-model-with-nemo.ipynb

```diff
@@ -0,0 +1,871 @@
+    "\n",
+    "This tutorial covers fine-tuning only the NMT Multilingual model. Fine-tuning a Multilingual model is a relatively more challenging task (like choosing a balanced dataset covering multiple languages). At this stage, multilingual fine-tuning is only supported with specific NeMo and Pytorch lightning versions(PTL<2.0). We suggest you to use the specific NeMo branch as shared here.\n",
+    "\n",
+    "The process of fine-tuning here can be split into following steps:\n",
+    "1. Data download.\n",
+    "2. Data preprocessing.\n",
+    "3. Fine-tuning the NMT model with NeMo.\n",
+    "4. Evaluate the fine-tuned NMT model with NeMo.\n",
+    "5. Exporting the NeMo model\n",
+    "6. Deploying the fine-tuned NeMo NMT model on the Riva Speech Skills server.  \n",
+    "\n",
+    "Let's walk through each of these steps in detail."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "95eb024b",
+   "metadata": {},
+   "source": [
+    "<a id='nmt_requirements_and_setup'></a>\n",
+    "### Requirements and Setup\n",
+    "\n",
+    "This tutorial needs to be run from inside a NeMo docker container. If you are not running this tutorial through a NeMo docker container, please refer to the [Riva NMT Tutorials](https://ngc.nvidia.com/resources/riem1phmzvud:riva:riva_nmt_ea_tutorials)'s [README.md](https://ngc.nvidia.com/resources/riem1phmzvud:riva:riva_nmt_ea_tutorials/files?version=2.2.0-ea) to get started.\n",
+    "\n",
+    "Before we get into the Requirements and Setup, let us create a base directory for our work here. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b60c3306",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "base_dir = \"NMTFinetuning\"\n",
+    "!mkdir $base_dir\n",
+    "base_dir=os.path.abspath(\"NMTFinetuning\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f5058992",
+   "metadata": {},
+   "source": [
+    "1. Clone the [NeMo github repository](https://github.com/NVIDIA/NeMo)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "id": "b80c6716",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cloning into '/mount/workspace/NMTFinetuning/NeMo2'...\n",
+      "remote: Enumerating objects: 75109, done.\u001b[K\n",
+      "remote: Counting objects: 100% (144/144), done.\u001b[K\n",
+      "remote: Compressing objects: 100% (86/86), done.\u001b[K\n",
+      "remote: Total 75109 (delta 66), reused 83 (delta 58), pack-reused 74965\u001b[K\n",
+      "Receiving objects: 100% (75109/75109), 134.73 MiB | 43.88 MiB/s, done.\n",
+      "Resolving deltas: 100% (54507/54507), done.\n",
+      "Updating files: 100% (1875/1875), done.\n"
+     ]
+    }
+   ],
+   "source": [
+    "NeMoBranch = \"r1.19.0\"\n",
+    "!git clone -b $NeMoBranch https://github.com/bpritam14/NeMo.git $base_dir/NeMo"
```

Comment:
this is not cloning from NVIDIA/nemo ?

Reply:
We changed this because finetuning works with specific version of NeMo and also required some changes. The latest NeMo container (PTL >=2.0) doesn't support multilingual fine-tuning.